irohad/multi_sig_transactions/transport/impl/mst_transport_grpc.cpp

```diff
@@ -87,19 +86,9 @@ grpc::Status MstTransportGrpc::SendState(
   async_call_->log_->info("batches in MstState: {}",
                           new_state.getBatches().size());
 
-  auto &peer = request->peer();
-  factory_
-      ->createPeer(peer.address(),
-                   shared_model::crypto::PublicKey(peer.peer_key()))
-      .match(
-          [&](expected::Value<std::unique_ptr<shared_model::interface::Peer>>
-                  &v) {
-            subscriber_.lock()->onNewState(std::move(v.value),
-                                           std::move(new_state));
-          },
-          [&](expected::Error<std::string> &e) {
-            async_call_->log_->info(e.error);
-          });
+  subscriber_.lock()->onNewState(
+      shared_model::crypto::PublicKey(request->source_peer_key()),
```

Comment:
`request->source_peer_key()` should be validated before creating a shared_model object.